src/qtcore/qml/qml.js

```diff
@@ -360,6 +363,43 @@ function applyProperties(metaObject, item, objectScope, componentScope) {
                         throw "Cannot set alias property pointing to an QML object.";
                     this.componentScope[this.val.objectName].$properties[this.val.propertyName].set(newVal, reason, objectScope, componentScope);
                 }
+
+                if (value.propertyName) {
+                  var con = function(prop) {
+                    var obj = prop.componentScope[prop.val.objectName];
+                    if (!obj) {
+                      console.error("qtcore: target object ",prop.val.objectName," not found for alias ",prop );
+                    } else {
+                      var targetProp = obj.$properties[prop.val.propertyName];
+                      if (!targetProp) {
+                        console.error("qtcore: target property [",prop.val.objectName,"].",prop.val.propertyName," not found for alias ",prop.name );
+                      } else {
+                        // targetProp.changed.connect( prop.changed );
+                        // it is not sufficient to connect to `changed` of source property
+                        // we have to propagate own changed to it too
+                        // seems the best way to do this is to make them identical?..
+                        // prop.changed = targetProp.changed;
+                        // obj[i + "Changed"] = prop.changed;
+                        // no. because those object might be destroyed later.
+                        ( function() {
+                          var loopWatchdog = false;
+                          targetProp.changed.connect( item, function() {
+                              if (loopWatchdog) return; loopWatchdog = true;
+                              prop.changed.apply( item,arguments );
+                              loopWatchdog = false;
+                          } );
+                          prop.changed.connect( obj, function() {
+                              if (loopWatchdog) return; loopWatchdog = true;
+                              targetProp.changed.apply( obj, arguments );
+                              loopWatchdog = false;
+                          } );
+                        } ) ();
```

Comment:
Note: code style is not perfect here, e.g. `(function() {` wrap is not needed. But we can fix that later at the cleanup stage, I would want to keep it like this until we complete the merges.